### ISSUE TITLE
unknown directive "set_by_lua"

### ISSUE THREAD

**Issue by phoen1x** [Mailu/Mailu#227]
Could you please check the stable branch. There seems to a bug related to a [cherry pick](https://github.com/Mailu/Mailu/issues/225#issuecomment-319162533) from issue #225 

After upgrading my instance I ran into 
 ```
http_1       | nginx: [emerg] unknown directive "set_by_lua" in /etc/nginx/nginx.conf:46
 ```
which seems to be cause by the missing directive
 ```
include /etc/nginx/modules/http_lua.conf;
 ```
Also see file history of nginx.conf.default

* [Properly load the nginx lua module](https://github.com/Mailu/Mailu/commit/4bdd8c1df5fc29e9cdd350aaaadd05daa405af73)

I fixed my instance by cherry picking the following files from the master branch

* [nginx.conf.default](https://github.com/Mailu/Mailu/blob/master/nginx/nginx.conf.default)
* [nginx.conf.fallback](https://github.com/Mailu/Mailu/blob/master/nginx/nginx.conf.fallback)
* [start.sh](https://github.com/Mailu/Mailu/blob/master/nginx/start.sh)

**Comment by kaiyou** [comment]
Thank you for reporting, I missed a commit indeed when cherry picking.

**Comment by kaiyou** [comment]
Just cherry picked the missing include, could you test once the build is finished?

**Comment by phoen1x** [comment]
After pulling the changes and rebuilding my custom Docker containers everything seems to work fine. THX for the quick fix.

I leave this issue open because there seems to be another bug in the start.sh of the [stable branch](https://github.com/Mailu/Mailu/blob/stable/nginx/start.sh#L3)
```bash
if [[ -z ENABLE_CERTBOT || -f /certs/cert.pem ]]
```
I saw a missing $ before ENABLE_CERTBOT and Atom complains about a [POSIX error](https://github.com/koalaman/shellcheck/wiki/SC2039)

the [master branch](https://github.com/Mailu/Mailu/blob/master/nginx/start.sh#L3) looks like that
```bash
if [ -z $ENABLE_CERTBOT ] || [ -f /certs/cert.pem ]
```

A quick Test with
```bash
#!/bin/sh

if [[ -z ENABLE_CERTBOT ]]
then
    echo "1 unset"
else
    echo "1 set"
fi

if [ -z ENABLE_CERTBOT ]
then
    echo "2 unset"
else
    echo "2 set"
fi


if [ -z $ENABLE_CERTBOT ]
then
    echo "3 unset"
else
    echo "3 set"
fi
```

shows wrong results in test 1 and 2

```bash
linux$ echo $ENABLE_CERTBOT

linux$ ./test.sh
./test.sh: 3: ./test.sh: [[: not found
1 set
2 set
3 unset
```

**Comment by phoen1x** [comment]
I'm closing this issue due to @dosimeter reference. The addressed error **unknown directive "set_by_lua"** is fixed. The minor bug mentioned [here](https://github.com/Mailu/Mailu/issues/227#issuecomment-320468189) should be addressed in another issue.
